xarray/core/indexing.py

```diff
@@ -253,9 +261,31 @@ def remap_label_indexers(data_obj, indexers, method=None, tolerance=None):
 
     dim_indexers = get_dim_indexers(data_obj, indexers)
     for dim, label in dim_indexers.items():
-        try:
+        if dim in data_obj.indexes:
             index = data_obj.indexes[dim]
-        except KeyError:
+
+            coords_dtype = data_obj.coords[dim].dtype
+            label = maybe_cast_to_coords_dtype(label, coords_dtype)
+            idxr, new_idx = convert_label_indexer(index, label, dim,
+                                                  method, tolerance)
+            pos_indexers[dim] = idxr
+            if new_idx is not None:
+                new_indexes[dim] = new_idx
+
+        elif dim in data_obj.coords and len(data_obj.coords[dim] == 1):
```

Comment:
I think this should be `data_obj.coords[dim].ndim == 1` instead of `len(data_obj.coords[dim] == 1)`? The later is checking that the coordinate's first dimension has length 1.